git-imerge

```diff
@@ -3628,19 +3628,63 @@ def cmd_diagram(parser, options):
         )
 
 
+def reparent_recursively(git, start_commit, parents, end_commit):
```

Comment:
nit: this isn't a recursive process but rather an iterative process

Reply:
It is in fact recursive; it's just that the recursion is built into `git log --topo-order --reverse` rather than coded explicitly here. But this is just an internal name so I don't think the distinction matters much.

Reply:
Make sense.